### PR TITLE
Clone printf and fprintf calls

### LINKED ISSUE #1146: Support the use of printf statements in the original function
These statements should be identified and cloned successfully in Clad

### PR DESCRIPTION
**Pull request by kchristin22**
Closes #1146

### CODE REVIEW

File: lib/Differentiator/ReverseModeVisitor.cpp [hunk 1]
```diff
@@ -1456,6 +1456,9 @@ Expr* getArraySizeExpr(const ArrayType* AT, ASTContext& context,
       return StmtDiff(Clone(CE));
     }
 
+    if (FD->getNameAsString() == "printf" || FD->getNameAsString() == "fprintf")
```
Comment (vgvassilev) [inline lib/Differentiator/ReverseModeVisitor.cpp:1459]: Can we add a fixme to revisit after the variadic pullbacks issue is closed?

File: test/CUDA/GradientKernels.cu [hunk 2]
```diff
@@ -520,6 +521,7 @@ void launch_add_kernel_4(int *out, int *in, const int N) {
 
 // CHECK: void launch_add_kernel_4_grad_0_1(int *out, int *in, const int N, int *_d_out, int *_d_in) {
 //CHECK-NEXT:    int _d_N = 0;
+//CHECK-NEXT:    printf("Launching add_kernel_4 for size: %d\n", N);
```
Comment (vgvassilev) [inline test/CUDA/GradientKernels.cu:524]: Can we add a test where printf is in a loop printing the loop variable value?

File: lib/Differentiator/ReverseModeVisitor.cpp (hunk 1 above)
Comment (kchristin22) [reply lib/Differentiator/ReverseModeVisitor.cpp:1460]: Added

File: test/CUDA/GradientKernels.cu (hunk 2 above)
Comment (kchristin22) [reply test/CUDA/GradientKernels.cu]: Done as well

**Comment by kchristin22** [comment]
This PR will be revisited when #1150 is fixed.

**Pull request merged**